airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cursor_based/CursorBasedCtidUtils.java

```diff
@@ -80,12 +81,18 @@ public static StreamsCategorised<CursorBasedStreams> categoriseStreams(final Sta
 
     final List<ConfiguredAirbyteStream> newlyAddedStreams = identifyNewlyAddedStreams(fullCatalog, alreadySeenStreamPairs);
     final List<ConfiguredAirbyteStream> streamsForCtidSync = getStreamsFromStreamPairs(fullCatalog, stillInCtidStreamPairs);
-    streamsForCtidSync.addAll(newlyAddedStreams);
-
-    final List<ConfiguredAirbyteStream> streamsForCursorBasedSync = getStreamsFromStreamPairs(fullCatalog, cursorBasedSyncStreamPairs);
+    final List<ConfiguredAirbyteStream> streamsForCursorBasedOrFullRefreshSync = getStreamsFromStreamPairs(fullCatalog, cursorBasedSyncStreamPairs);
```

Comment:
Can we just modify streams from stream pairs to return streams that are only in incremental mode?

IIUC the issue here is that a full refresh mode follows a different path and full refresh is handled by another path, so I'm think we don't need to specifically care about those streams here. `AbstractDbSource.getFullRefreshIterators` and we are not filtering out the full refresh streams here?

Reply:
The FullRefresh path is actually handled when the stream is passed to `super.getIncrementalIterators`. So we would need to know which stream from `newlyAddedStreams` is FullRefresh or not.

Reply:
That's not quite correct. We have a seperate codepath in AbstractDbSource which explicitly handles streams in full refresh. getIncrementalIterators should only handle incremental streams and filter out all full refresh. https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/AbstractDbSource.java#L164

Reply:
I see what you're saying.
the code in `AbstractDBSource.read()` only expects us to only return iterators that are incremental. 
Full refresh is handled afterwards
```
    final List<AutoCloseableIterator<AirbyteMessage>> incrementalIterators =
        getIncrementalIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager,
            emittedAt);
    final List<AutoCloseableIterator<AirbyteMessage>> fullRefreshIterators =
        getFullRefreshIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager,
            emittedAt);
```

Reply:
I think we want to apply the same filter to getStreamsFromStreamPairs too, correct? And I think we still shouldn't care about full refresh here (and in the XminCtidUtils) right? `streamsForCursorBasedOrFullRefreshSync`

Reply:
Added